library(readxl)
library(reshape2)
library(ggplot2)

path = "/icgc/dkfzlsdf/analysis/B080/sharma/boulantLab/bulk_hypoxia/"

# Load count data generated by feature counts
dat = readRDS(paste0(path, "/data/counts/hypoxia_counts.RDS"))

# Save FeatureCounts summary for MultiQC
tmp = dat$stat
colnames(tmp) = gsub(".", "_", colnames(tmp), fixed=T)
colnames(tmp) = gsub("_1_1", "-1-1", colnames(tmp), fixed=T)
colnames(tmp) = gsub("_bam", ".bam", colnames(tmp), fixed=T)
write.table(tmp, file = paste0(path, "/data/counts/hypoxia_counts.summary"), quote=F, row.names = F, sep="\t")
rm(tmp)

# Save Rsubreads summary for MultiQC
# f = list.files(paste0(path,"data/bam"), pattern = ".summary", full.names = T)
# tmp = lapply(f, function(x) read.table(x))
# tmp2 = sapply(tmp, function(x) x[,2])
# colnames(tmp2) = basename(f)
# rownames(tmp2) = tmp[[1]][,1]
# tmp2 = t(tmp2)
# 
# tmp = round(tmp2/tmp2[,1] * 100, 2)
# tmp[,1] = tmp2[,1]
# tmp = melt(tmp)
# rm(tmp2, f)
# 
# ggviolin(tmp, x = "dose", y = "len",
#          add = "dotplot")

# Load sample annotation
anno = as.data.frame(read_excel(paste0(path, "data/anno/carmon_hypoxia_metadata.xlsx")))

# Restrict to protein coding genes only
pcg = dat$annotation[dat$annotation$gene_biotype == "protein_coding",]
cnt = dat$counts[pcg$GeneID,]

# Restrict to genes whose row sum > 0
cnt = cnt[rowSums(cnt) > 0,]
pcg = pcg[pcg$GeneID %in% rownames(cnt),]

# Sanity check
identical(rownames(cnt), pcg$GeneID)
#[1] TRUE

# Simplify column names using the samples annotation file
colN = gsub(".bam", "", colnames(cnt))
fqN  = gsub("-", ".", gsub("_", ".", gsub("_fastq.gz", "", anno$Samples)))

cnt = cnt[, match(fqN, colN)]

# Sanity check
colN = gsub(".bam", "", colnames(cnt))
fqN  = gsub("-", ".", gsub("_", ".", gsub("_fastq.gz", "", anno$Samples)))

identical(colN, fqN)
# TRUE

colnames(cnt) = anno$SampleID

# Additional time annotation
anno$Time2[anno$Time %in% c("6h", "12h")] = "Early"
anno$Time2[anno$Time %in% c("D1", "D2")] = "Late"
anno$Time2 = factor(anno$Time2, levels = c("Early", "Late"))

# Additional type annotation
anno$Type2 = paste(anno$Time2, 
                   sapply(strsplit(as.character(anno$Type), ""), 
                          function(x)x[1]), sep="_")
anno$Type2 = factor(anno$Type2, levels = c("Early_H", "Early_N", "Late_H", "Late_N"))

# Simplifying sample annotation file
anno$Time = factor(anno$Time, levels = c("6h", "12h", "D1", "D2"))
anno$Type = factor(anno$Type, levels = c("Hypoxic", "Normoxic"))
anno$Condition = factor(anno$Condition, levels = c("H_6h", "N_6h", "H_12h", "N_12h", "H_D1", "N_D1", "H_D2", "N_D2"))
anno = data.frame(anno[,c(1, 3:ncol(anno))], row.names = 1)

# Saving results for downstream analysis using DEseq2
saveRDS(list(counts = cnt, geneanno = pcg, sampanno = anno),
        paste0(path, "data/counts/hypoxia_filtered_counts.RDS"))

write.csv(data.frame(Gene = rownames(cnt), cnt, stringsAsFactors = F),
          paste0(path, "data/counts/hypoxia_filtered_counts.csv"), 
          quote = FALSE, row.names = F)
